src/components/typeahead/typeahead.component.ts

```diff
@@ -215,6 +255,20 @@ export class TypeaheadComponent<T = any> implements OnChanges, OnDestroy {
 
     optionClickHandler(_event: MouseEvent, option: TypeaheadVisibleOption<T>): void {
         this.select(option, 'mouse');
+
+        if (Array.isArray(this.recentOptions)) {
+            // If the option is already in the array, remove it first.
+            let index = this.recentOptions.indexOf(option.value);
+            if (index != -1) {

```

Comment:
Can we use `!==` as we always want to do value and type checking

Reply:
Done.